File: src/main/java/me/syari/ss/discord/api/entities/Member.java
package me.syari.ss.discord.api.entities;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;

public interface Member extends IMentionable {

    @Nonnull
    User getUser();


    @Nullable
    String getNickname();


    @Nonnull
    String getDisplayName();


}

File: src/main/kotlin/me/syari/ss/discord/entities/Member.kt
package me.syari.ss.discord.entities

data class Member(
    val guild: Guild, val user: User
): WithId {
    var nickname: String? = null

    val displayName
        get() = (if (nickname != null) nickname else user.name) ?: "null"

    override val idLong
        get() = user.idLong
}